CityOfZion/neo-sharp
# Repository refactor (#122)

* refactor UtNetworkManager

* * Asserting / AssertingThat fluent API
* Move Max and Min calculation from the Fixed8 to extension methods
* Add +50% test coverage to the Fixed8 structure
* fix small compilation issue in UtNetworkManager

* Fixed8 tests almost 100%

* refactor UtProtocolV2 tests in order to be readable, using the Async/Await pattern and other test patterns used already in other tests.

* Replace the AssertingThat fluent API to FluentAssertions

* fix some typo mistakes

* UtBootstrapper tests

* compilation error fixed

* * Fix unstable test in UtServer. Replace Delay usage with AutoResetEvent
* Replace static calls with extension methods
* Add "this" keyword
* Rename the test with most meaning name.

* Review UtServer Tests
* Add ILoggerProvider interface to be able to test it.

* Refactor some tests in UtServer

* Add tests for the logic in ListenForMessages private method.
test coverage is up to 91%. Missing test for exception cases.

* now it's possible to verify the LowWarning message.

* add UtNetworkAclFactory

* Remove dependency of IServer in VersionMessageHandler.

* Remove IServer dependency from VersionMEssageHandler and replace with IServerContext

* Repository pattern on the persistency layer

* clean up the usings
